eth/abc.py

```diff
@@ -1573,6 +1573,16 @@ def commit(self, checkpoint: JournalDBCheckpoint) -> None:
         """
         ...
 
+    @abstractmethod
+    def lock_changes(self) -> None:
+        """
+        Locks in changes to storage, typically just as a transaction starts.
+
+        This is used, for example, to look up the storage value from the start
```

Comment:
Alternative: "This is used to ensure that reads which skip the journal read state as of the beginning of the transaction, rather than state as of the beginning of the block"